### PR TITLE
Added dockblock to ComponentManager class

### PR DESCRIPTION
**Pull request by ahcorde**
Added dockblock to ComponentManager class

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rclcpp_components/include/rclcpp_components/component_manager.hpp [hunk 1]
```diff
@@ -60,41 +62,89 @@ class ComponentManager : public rclcpp::Node
...
+   * \param node_name the name of the node that the data originates from.
+   * \param node_options additional options to control creation of the node.
+   */
   RCLCPP_COMPONENTS_PUBLIC
   ComponentManager(
     std::weak_ptr<rclcpp::Executor> executor,
     std::string node_name = "ComponentManager",
     const rclcpp::NodeOptions & node_options = rclcpp::NodeOptions());
 
+  /// Default destructor
```
Comment (brawner) [inline rclcpp_components/include/rclcpp_components/component_manager.hpp:80]: Not needed.

File: rclcpp_components/include/rclcpp_components/component_manager.hpp [hunk 2]
```diff
@@ -60,41 +62,89 @@ class ComponentManager : public rclcpp::Node
...
   RCLCPP_COMPONENTS_PUBLIC
   virtual ~ComponentManager();
 
-  /// Return a list of valid loadable components in a given package.
+  /// Get the component resource
```
Comment (brawner) [inline rclcpp_components/include/rclcpp_components/component_manager.hpp:84]: The other comment had more information, why the change?

File: rclcpp_components/include/rclcpp_components/component_manager.hpp [hunk 3]
```diff
@@ -60,41 +62,89 @@ class ComponentManager : public rclcpp::Node
...
+   * \throws ComponentManagerException if the resource was not found or a invalid resource entry
+   * \return a list of component resources
+   */
   RCLCPP_COMPONENTS_PUBLIC
   virtual std::vector<ComponentResource>
   get_component_resources(
     const std::string & package_name,
     const std::string & resource_index = "rclcpp_components") const;
 
+  /// Create the component factory
```
Comment (brawner) [inline rclcpp_components/include/rclcpp_components/component_manager.hpp:97]: Is there more detail you can add to the brief?

File: rclcpp_components/include/rclcpp_components/component_manager.hpp [hunk 4]
```diff
@@ -60,41 +62,89 @@ class ComponentManager : public rclcpp::Node
...
+  /*
+   * \param resource list of component resources
+   */
   RCLCPP_COMPONENTS_PUBLIC
   virtual std::shared_ptr<rclcpp_components::NodeFactory>
   create_component_factory(const ComponentResource & resource);
 
 protected:
+  // Service to load a new node in the component
```
Comment (brawner) [inline rclcpp_components/include/rclcpp_components/component_manager.hpp:106]: This is the service callback, right? Say that instead. Also, when might a deriving class want to override these functions?

These questions apply also to the next two functions.

Comment (ahcorde) [reply rclcpp_components/include/rclcpp_components/component_manager.hpp]: If you derive from this class and change these methods we are going to change the behaviour of this abstraction. do you think we should explicitly indicate it?

Comment (brawner) [reply rclcpp_components/include/rclcpp_components/component_manager.hpp]: I was just surprised to see protected virtual callbacks, and I wasn't entirely sure when that would be useful to override, so I thought some documentation might help. But seeing as all the public functions are virtual as well, I guess this class was just designed with that sort of flexibility in mind.

**Comment by ahcorde** [comment]
running CI up-to `rclcpp_components` to check linters

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10717/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6130/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8732/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10617/)
